File: LeetCode/Java Solutions/Remove K Digits/remove.java
public class Solution {
    public String removeKdigits(String num, int k) {
        StringBuilder sb = new StringBuilder();
        for (char digit : num.toCharArray()) {
            while (k > 0 && sb.length() > 0 && sb.charAt(sb.length() - 1) > digit) {
                sb.deleteCharAt(sb.length() - 1);
                k--;
            }
            sb.append(digit);
        }
        
        // Remove the last k digits if k is still greater than 0
        while (k > 0) {
            sb.deleteCharAt(sb.length() - 1);
            k--;
        }
        
        // Remove leading zeros
        while (sb.length() > 1 && sb.charAt(0) == '0') {
            sb.deleteCharAt(0);
        }
        
        return sb.length() == 0 ? "0" : sb.toString();
    }
}

File: LeetCode/Python Solutions/Remove K Digits/remove.py
class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        stack = []
        for digit in num:
            while k > 0 and stack and stack[-1] > digit:
                stack.pop()
                k -= 1
            stack.append(digit)
        
        # If k is still greater than 0, remove the last k digits
        final_stack = stack[:-k] if k else stack
        
        # Convert stack to string and strip leading zeros
        return ''.join(final_stack).lstrip('0') or "0"